spec/support/transactions.rb

```diff
@@ -175,6 +178,21 @@ def event_subscriber
       def run
         test_client.subscribe(Mongo::Monitoring::COMMAND, event_subscriber)
 
+        $distinct_ran ||= if @ops.any? { |op| op.name == 'distinct' }
```

Comment:
What's different about distinct that requires special handling?

Reply:
This behavior is mandated by the spec.

Reply:
Can you share a link to where this is written? From looking at RUBY-1792, the only spec commit mentioned in the linked drivers ticket is https://github.com/mongodb/specifications/commit/2d2003825f2a11f360440d1b77371cff8370bf82, and I can't find anything there about distinct.

Reply:
This requirement is described in https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#why-do-tests-that-run-distinct-sometimes-fail-with-staledbversion.